src/karthik/narcissisticNumbers/narcissisticNumbers.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Function to check if number is naricissistic or not.
+ * @param {number} num - given number to check
+ * @return {boolean} true or false
+ */
+const checkIfNarcissticNumber = (num: number): boolean => {
+  if (num < 0) return false;
+  const arr = ('' + num).split('');
+  const noOfDigits = arr.length;
+
+  let sum = 0;
+  for (let i = 0; i < noOfDigits; i++) {
+    sum += parseInt(arr[i]) ** noOfDigits;
+  }
+
+  if (num === sum) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+/**
+ * Function to Find all numbers in range if they are Naricissistic in value
+ * @param {number} from - starting value of the range
+ * @param {number} to - ending value of the range
+ * @return {number[]} - Array of Naricissistic Numbers
+ */
+const getNarcissticNumbersWithinRange = (
+  from: number,
+  to: number
+): Array<number> => {
+  const naricissisticNumbers: Array<number> = [];
+  for (let checkNum = from; checkNum <= to; checkNum++) {
+    const checkIfNaricissistic: boolean = checkIfNarcissticNumber(checkNum);
+    if (checkIfNaricissistic) {
```

Comment:
We can directly call checkIfNarcissticNumber(checkNum) inside if, as variable checkIfNaricissistic is used at once only.

Reply:
I wrote that seperately for readability.

If i include the function directly in if() I would not know the returning value data type.

Person new to the code will have to check the function return data type or know the truthy and falsy concepts exactly to understand the return value.

Reply:
Typescript will make the types flow along with the code, If the return type is defined for a function we don't need to define type for the variable it got assigned and type of the variable can be seen by hovering on it.
Also I don't see a need to add extra lines just to find the type.

Reply:
Ok I got it. Added the changes now.